messaging/src/main/java/org/axonframework/serialization/avro/AvroConverter.java

```diff
@@ -0,0 +1,241 @@
+            return false; // currently only support types which are classes
+        }
+    }
+
+    @Nullable
+    @Override
+    public <T> T convert(@Nullable Object input, @Nonnull Type targetType) {
+        if (input == null) {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Input to convert is null, so returning null immediately.");
+            }
+            return null;
+        }
+
+        if (!(targetType instanceof Class<?> targetClass)) {
+            throw new ConversionException("Cannot convert input to " + targetType + ", target type must be a class.");
+        }
+
+        Class<?> sourceClass = input.getClass();
+        if (sourceClass.equals(targetClass) || targetClass.isAssignableFrom(sourceClass)) {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Casting given input since source and target type are identical.");
+            }
+            //noinspection unchecked
+            return (T) input;
+        }
+
+        if (strategyForType(sourceClass)) { // for serialization
+            // input must be supported by the strategy
+            // target is either byte[].class orr can be converted to it by the chain
+            if (byte[].class.equals(targetType)) {
+                return serializeByStrategy(input, sourceClass);
+            } else if (converter.canConvert(byte[].class, targetType)) {
+                return converter.convert(serializeByStrategy(input, sourceClass), targetType);
+            } else {
+                throw new ConversionException("Cannot serialize given input to target type [" + targetType + "]");
+            }
+        } else if (strategyForType(targetClass)) { // for de-serialization
+
+            if (byte[].class.equals(sourceClass)) {
+                //   straight from the source:
+                //     input is byte[].class
+                //     target type must be supported by the strategy
+                return deserializeByStrategy((byte[]) input, targetClass);
+            } else if (converter.canConvert(sourceClass, byte[].class)) {
+                //   straight from the source:
+                //     input is something converted to byte[].class by the chain
+                //     target type must be supported by the strategy
+
+                //noinspection DataFlowIssue
+                return deserializeByStrategy(converter.convert(input, byte[].class), targetClass);
+            } else if (GenericRecord.class.isAssignableFrom(sourceClass)) {
+                //   through upcaster / intermediate representation:
+                //     2. after upcaster
+                //      input is GenericRecord (upcasted)
+                //      target type must be supported by the strategy
+                return deserializeByStrategy((GenericRecord) input, targetClass);
+            } else {
+                throw new ConversionException("Cannot deserialize given input to target type [" + targetType + "]");
```

Comment:
```suggestion
                throw new ConversionException("Cannot convert given input to target type [" + targetType + "]");
```

Reply:
Because of consistency. We have decided to replace **All** serialization logic in favor of conversion logic. With that decision, to me, comes the replacement of **All** mentions of serialization. That thus includes verbs like "serialize" and "deserialize." 

It is a nit, for sure, but having "convert" in some places and "serialize" in others makes Axon Framework ambiguous.